internal/services/service_test.go

```diff
@@ -20,108 +19,90 @@ func setup() {
 	buildFakeService()
 }
 
+const (
+	servicePort   = "55555"
+	serviceAdress = "0.0.0.0:" + servicePort
+)
+
 func TestService(t *testing.T) {
 	setup()
 
 	t.Run("it stops when the context is cancelled", func(t *testing.T) {
 		service := Service{
-			name:       "fake-service",
-			binaryName: "fake-service",
+			name:            "fake-service",
+			binaryName:      "fake-service",
+			healthcheckPort: servicePort,
 		}
 		ctx, cancel := context.WithCancel(context.Background())
 		defer cancel()
 
-		startErr := make(chan error)
-		success := make(chan struct{})
+		// start service in goroutine
```

Comment:
I simplified all the test cases quite a bit. I think I maintained the same behavior.